src/sentry/issues/status_change_consumer.py

```diff
@@ -137,40 +137,52 @@ def update_status(group: Group, status_change: StatusChangeMessageData) -> None:
         )
 
 
+def get_group_from_fingerprint(project_id: int, fingerprint: Sequence[str]) -> Group | None:
+    results = bulk_get_groups_from_fingerprints([(project_id, fingerprint)])
+    return results.get((project_id, tuple(fingerprint)))
+
+
 def bulk_get_groups_from_fingerprints(
     project_fingerprint_pairs: Iterable[tuple[int, Sequence[str]]]
-) -> dict[tuple[int, str], Group]:
+) -> dict[tuple[int, tuple[str, ...]], Group]:
     """
     Returns a map of (project, fingerprint) to the group.
 
     Note that fingerprints for issue platform are expected to be
     processed via `process_occurrence_data` prior to calling this function.
     """
     fingerprints_by_project: dict[int, list[str]] = defaultdict(list)
-    for project_id, fingerprints in project_fingerprint_pairs:
-        fingerprints_by_project[project_id].append(fingerprints[0])
+    for project_id, hashes in project_fingerprint_pairs:
+        fingerprints_by_project[project_id].extend(hashes)
 
     query = GroupHash.objects.none()
-    for project_id, fingerprints in fingerprints_by_project.items():
+    for project_id, hashes in fingerprints_by_project.items():
         query = query.union(
             GroupHash.objects.filter(
                 project=project_id,
-                hash__in=fingerprints,
+                hash__in=hashes,
             ).select_related("group")
         )
 
-    result = {
+    group_hash_result = {
         (grouphash.project_id, grouphash.hash): grouphash.group
         for grouphash in query
         if grouphash.group is not None
     }
+    result = {}
+    for project_id, fingerprint in project_fingerprint_pairs:
+        for hash in fingerprint:
+            # See if we managed to load any of the hashes from the fingerprint, prioritising early ones first
```

Comment:
what does "early" mean here?

Reply:
Probably not worded well - fingerprints earlier in the list have higher priority. I'll reword this